### DISCUSSION TITLE
Media fragments not supported in Thorium?

### DISCUSSION THREAD

**Discussion by koechsle** [edrlab/thorium-reader#1961] [General]
Hi it seems thorium v 2.2.0 (on windows 11) doesn't understand media fragments? They have the format sound.mp3#t=NN

The following LPF file into Thorium with a publication.json file and the specified mp3 files opens the file all right and the index is shown but there is no sound player - just a blank window.

publication.json content:
{
"@context":["https://schema.org","https://www.w3.org/ns/pub-context", {"language":"da-DK"}],
"conformsTo":"https://www.w3.org/TR/audiobooks/",
"type": "Audiobook",
"id"  : "urn:isbn:XXX",
"name":"test",
"accessMode": "audio",
"author": "Kim Oechsle",
"readBy": "Kim Oechsle",
"publisher": "Orange Ink",
"url": "https://mysite.com",
"readingOrder":[
    {"url":"001-forside+dedikation.mp3#t=0", "encodingFormat": "audio/mpeg","name":"DU ER en bevidsthed i EN HJERNE I EN KROP"},
    {"url":"002-indhold.mp3#t=0","encodingFormat": "audio/mpeg","name":"Indhold"}
]
}

It works fine if I change it to

{
"@context":["https://schema.org","https://www.w3.org/ns/pub-context", {"language":"da-DK"}],
"conformsTo":"https://www.w3.org/TR/audiobooks/",
"type": "Audiobook",
"id"  : "urn:isbn:XXX",
"name":"DU ER en bevidsthed i EN HJERNE I EN KROP",
"accessMode": "audio",
"author": "Kim Oechsle",
"readBy": "Kim Oechsle",
"publisher": "Orange Ink",
"url": "https://mysite.com",
"readingOrder":[
    {"url":"001-forside+dedikation.mp3","duration": "PT00H00M38S", "encodingFormat": "audio/mpeg","name":"DU ER en bevidsthed i EN HJERNE I EN KROP"},
    {"url":"002-indhold.mp3","duration": "PT00H02M24S","encodingFormat": "audio/mpeg","name":"Indhold"}
]
}

The trouble for me is that I have a 9h long mp3 file which is indexed into A LOT of small fragments.
The W3 Audiobook format accepts media fragments.

Are there any plans on adding this feature to Thorium?

Best regards,

Kim

**Comment by danielweck** [comment]
Hello, you are correct I am afraid. In the Readium web publication manifest format we decided not to support media fragment at the level of "raw" resources (items in the reading order effectively are). Media fragments such as audio "start time" are of course supported in the table of contents (much like with text references `chapter1.html#section2` we can have `chapter1.mp3#t=5`).

**Comment by danielweck** [comment]
I am moving this "issue" to a GitHub "discussion", as this is worth discussing indeed, but this is not an actionable issue at this point in time, especially given that introducing support for this W3C feature would require significant architectural changes in Thorium's current implementation.

Do you know implementations that support the authoring structure you are suggesting?

My personal analysis is that there are footguns in the W3C "audiobooks" specification that have not been adequately investigated when it comes to allowing fragments in reading order items, such clip-begin/end and duration mixed metadata, correlating TOC links that carry a start time with the correct reading order index, the added difficulty of validating such content, etc.

Note that the specification is just a draft at this point:

https://github.com/w3c/audiobooks/issues
https://w3c.github.io/audiobooks/

**Reply by koechsle** [reply]
Hi Daniel,

Thank you for responding immediately. I am new to Audiobook authoring. I have been using an Audiobook distribution tool where you upload a zip with mp3 files and then you can create a manifest file from the concatenated files by inputting chapters and timestamps.
My written book is supposed to be used as a reference so I wanted to do the same service to audiobook listeners: a comprehensive index with all chapters, subchapters and exercises. 
 
The resulting manifest looks like my first example, I just added a few optional values (but only included two timestamps). It uses media fragments and is supposed to live up to the W3C Audiobook specs. When I distribute the audiobook, the manifest is sent along to bookstores and they will then sell the audiobook in their own format, which I don't have control over.
The https://www.w3.org/TR/audiobooks/ is a Recommendation from 2020, not a draft (you linked to a 2023 draft version) so in my Audiobook naivety I was sure that this was the common way to navigate an audiobook!

Thorium was the tool I was recommended to use so I don't know a lot if other Audiobook readers implement the media fragment feature. I will try to find other tools and see if this is the case.
Thanks!
